### PR TITLE
deserialize: fix optional

### PR DESCRIPTION
**Pull request by jsign**
**Note: Merge #22 first.**

This PR fixes a bug when deserializing optional fields.

### CODE REVIEW

File: src/deserialize.zig [hunk 1]
```diff
@@ -183,7 +184,9 @@ pub fn deserialize(comptime T: type, allocator: Allocator, serialized: []const u
             if (serialized.len == 0 or serialized[0] == rlpByteListShortHeader or serialized[0] == rlpListLongHeader) {
                 out.* = null;
                 // 0 if serialized was empty, one in the case of an empty list
-                return serialized.len;
```
Comment (jsign) [inline src/deserialize.zig:185]: This line was incorrect. `serialized` is the full remaining bytes to process, not only the bytes scoped to the optional. 

This was found by trying to deserialize a tx sent to the zero address for creating a contract.

File: src/main.zig
Comment (gballet) [inline src/main.zig:1]: this looks like a merge error? I don't see the connection between optionals and `GenericRLPValue`?

**Review by gballet — CHANGES_REQUESTED** [review]
I think there was some branch issue, but the fix itself LGTM

File: src/main.zig
Comment (jsign) [reply src/main.zig:1]: As mentioned in the PR description:
> Note: Merge https://github.com/gballet/zig-rlp/pull/22 first.

This is why this PR is in draft.

Comment (jsign) [reply src/main.zig:1]: The reason is stacked on top, is that for [phant ](https://github.com/jsign/phant/pull/30) I need all the fixes in to make progress.

**Review by gballet — APPROVED** [review]
LGTM

**Pull request merged**